### PR TITLE
Support all integer types for dynamic dimensions

### PR DESCRIPTION
**Pull request by DennySun2020**

### CODE REVIEW

File: hatlib/function_info.py [hunk 1]
```diff
@@ -127,17 +138,31 @@ def verify(self, args: List[Any]):
 
         for i, (info, value) in enumerate(zip(self.arguments, args)):
             try:
-                if isinstance(value, np.ndarray) or isinstance(value, np.int64):
+                if (
+                    isinstance(value, np.ndarray)
+                    or isinstance(value, np.int64)
```
Comment (lisaong) [inline hatlib/function_info.py:143]: does something like this work?

`issubclass(type(value), np.integer)`

**Review by lisaong — APPROVED** [review]
added 1 minor comment

File: hatlib/function_info.py [hunk 2]
```diff
@@ -127,17 +138,31 @@ def verify(self, args: List[Any]):
...
+                    f"Error calling {self.name}(...): argument {i} failed verification: {v}"
+                )
 
     def as_cargs(self, args: List[Any]):
         "Converts arguments to their C interfaces"
         arg_values = [
-            ArgValue(info, value) if isinstance(value, np.ndarray) or isinstance(value, np.int64) else value
+            ArgValue(info, value)
+            if isinstance(value, np.ndarray)
+            or isinstance(value, np.int64)
```
Comment (lisaong) [inline hatlib/function_info.py:161]: same comment as above

File: hatlib/function_info.py (hunk 1 above)
Comment (DennySun2020) [reply hatlib/function_info.py:143]: It should work, much better.

**Pull request merged**